### ISSUE TITLE
Sync blocks on error retry

### ISSUE THREAD

**Issue by you06** [pingcap/tidb-binlog#866]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

Write data from upstream with insert/update DMLs and some DDLs.

```
TiDB -> pump -> drainer -> TiDB
```

- Mininal reproduce:

Prepare data

```sql
CREATE TABLE t(
  id int(11) NOT NULL AUTO_INCREMENT,
  a int(12) NOT NULL,
  PRIMARY KEY (id, a)
) PARTITION BY RANGE ( a ) (
  PARTITION p0 VALUES LESS THAN (1),
  PARTITION p1 VALUES LESS THAN (100),
  PARTITION pn VALUES LESS THAN (MAXVALUE)
);

INSERT INTO t(id, a) VALUES(1, 10), (2, 100), (3, 1000);
```

| Txn1 | Txn2 |
|-|-|
| BEGIN; | |
| UPDATE t SET a = 10000 WHERE id = 3; | |
| | ALTER TABLE t ADD COLUMN (c varchar(612) NOT NULL); |
| COMMIT; | |

2. What did you expect to see?

The DMLs and DDLs are synced successfully.

3. What did you see instead?

The sync blocks and there are many retry errors like `Error 1048: Column 'ggvub' cannot be null` in drainer.

```log
[2020/01/01 12:53:27.557 +00:00] [ERROR] [executor.go:85] ["exec fail"] [query="REPLACE INTO `pocket`.`fvmmmgz`(`fxrilmue`,`rdhddznuu`,`wfqsntb`,`cshvyv`,`zgbnm`,`qrllocl`,`id`,`qrpjh`,`swfcmcgo`,`unybkaklx`,`laimgr`,`wlbuuhkj`,`mnwjwbe`,`ggvub`) VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?)"] [args="[1453546922,\"U0Y4SVNSfk0kODB7KzU3N1FRV1JLVH01PTFIKypVWVh9QlNUQlNFKlhMSFpTKzFQLH45OFdGfkxVV1ZXJl03V19MQSM7O08=\",\"ViZUPX5XflYkVlAqOikme0xWTFEqJDdYVU5URTQ1XURZXVRUVF8rTUxRU1Q/U0xCUGBeRVshQ0FePig1VUh9QUMxIUQsQyQzWklde0VDfVEjNk1XUVFBNFRKX1A=\",\"Uk5fI0FMMGAxQDUvSSxGVUlRWjlYSjs3OVU4MSNLLFdGTUM/LFd8TVo9T0RDVEIjT086J0RFPk4lRntTXTVUfVAk\",\"WlpKJFRFPFVJWFd+RTtdR1RLJ0hJRkZGWVlGQy1MTCNUKkczREw=\",\"RVI+Sk1VUCdSSUtDQH5T\",30006,\"0000-00-00 00:00:00\",\"2034-03-11 00:08:48\",62558545,1931444655,\"Q0dKT0ZbIzI7fjZDWD4jK0A7XkVSI0dUWl5UTUdEIz9AQEc1NE5RSn0pIUs3VzhSPUdONEVdMUkxTEZ8NT9aX1c2Vj9ZXSRNTUU3VTUv\",\"Q1dNTVJDTEYpRkg2NUJHQTsjWiNaRkY0SFs5REZSIzddQU9dLCcuNDtEMTBLXygyfi0=\",null]"] [error="Error 1048: Column 'ggvub' cannot be null"]
```

4. Please provide the relate downstream type and version of drainer.
(run `drainer -V` in terminal to get drainer's version)

```
# /drainer -V
Release Version: v3.0.7-1-g28a4b0b
Git Commit Hash: 28a4b0bffc281cc55978cca68b312c9f664b02d8
Build TS: 2019-12-20 04:52:26
Go Version: go1.13.3
Go OS/Arch: linux/amd64
```

**Comment by suzaku** [comment]
How to reproduce this error?

**Comment by suzaku** [comment]
Related to pingcap/tidb#14235

**Comment by suzaku** [comment]
This issue is caused by a bug of the upstream database. The bug is already fixed in  pingcap/tidb#14235.
